File: apps/dsa-web/src/types/dailyReview.ts
export type ReviewDimension = 'day' | 'week' | 'month';

export interface DailyReviewAnomalyCounts {
  red: number;
  yellow: number;
}

export interface DailyReviewCharts {
  marketScores?: Record<string, number>;
  anomalyCounts?: DailyReviewAnomalyCounts;
}

export interface DailyReviewRegime {
  market: string;
  totalScore?: number;
  regime?: string;
  positionSuggestion?: string;
  reasoning?: string;
}

export interface DailyReviewAnomalyItem {
  level: string;
  name: string;
  message: string;
  action: string;
  affectedMarkets?: string[];
  possibleCause?: string;
  potentialImpact?: string;
}

export interface DailyReviewSnapshot {
  reviewDate?: string;
  generatedAt?: string;
  summary?: string;
  regimes?: DailyReviewRegime[];
  anomalies?: {
    redCount?: number;
    yellowCount?: number;
    items?: DailyReviewAnomalyItem[];
  };
  [key: string]: unknown;
}

export interface DailyReviewPeriodItem {
  dimension: ReviewDimension;
  periodKey: string;
  periodLabel: string;
  reviewDate: string;
  generatedAt?: string;
  summary: string;
  snapshot?: DailyReviewSnapshot;
  charts?: DailyReviewCharts;
}

export interface DailyReviewHistoryResponse {
  found: boolean;
  message: string;
  dimension: ReviewDimension;
  items: DailyReviewPeriodItem[];
}

export interface DailyReviewLatestResponse {
  found: boolean;
  message: string;
  dimension: ReviewDimension;
  item?: DailyReviewPeriodItem;
}

export interface DailyReviewRunRequest {
  pushTelegram?: boolean;
  useLlm?: boolean;
}

export interface DailyReviewRunResponse {
  success: boolean;
  message: string;
  item?: DailyReviewPeriodItem;
  reportPreview?: string;
}

export interface DailyReviewPushResponse {
  pushed: boolean;
  message: string;
}

File: api/v1/schemas/daily_review.py
# -*- coding: utf-8 -*-
"""Schemas for daily review APIs."""

from typing import Any, List, Literal, Optional

from pydantic import BaseModel, Field


ReviewDimension = Literal["day", "week", "month"]


class DailyReviewPeriodItem(BaseModel):
    """One review item for day/week/month dimension."""

    dimension: ReviewDimension = Field(...)
    period_key: str = Field(...)
    period_label: str = Field(...)
    review_date: str = Field("")
    generated_at: Optional[str] = Field(None)
    summary: str = Field("")
    snapshot: Optional[Any] = Field(None)
    charts: Optional[Any] = Field(None)


class DailyReviewHistoryResponse(BaseModel):
    """History response for daily review module."""

    found: bool = Field(...)
    message: str = Field("")
    dimension: ReviewDimension = Field(...)
    items: List[DailyReviewPeriodItem] = Field(default_factory=list)


class DailyReviewLatestResponse(BaseModel):
    """Latest period response for daily review module."""

    found: bool = Field(...)
    message: str = Field("")
    dimension: ReviewDimension = Field(...)
    item: Optional[DailyReviewPeriodItem] = Field(None)


class DailyReviewRunRequest(BaseModel):
    """Run request payload."""

    push_telegram: bool = Field(False, description="Whether to push report to Telegram after run")
    use_llm: bool = Field(False, description="Whether to use LLM summary when rerun")


class DailyReviewRunResponse(BaseModel):
    """Run response payload."""

    success: bool = Field(...)
    message: str = Field("")
    item: Optional[DailyReviewPeriodItem] = Field(None)
    report_preview: Optional[str] = Field(None)


class DailyReviewPushResponse(BaseModel):
    """Push-by-date response payload."""

    pushed: bool = Field(...)
    message: str = Field("")
